test/features/support/voucher_cancelation.js

```diff
@@ -0,0 +1,117 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { localhost } = require('./helpers/helpers');
+
+let specVoucherCancelation;
+let serialNumberParam;
+
+const baseUrl = `${localhost}vouchers/voucherstatuscheck/{voucherserialnumber}`;
+const requestFunction = () =>
+  specVoucherCancelation.patch(baseUrl).withBody({
+    'voucherserialnumber': `${serialNumberParam}`,
+    'Gov_Stack_BB': 'string'
+  }).withPathParams('voucherserialnumber', serialNumberParam);
+
+Before(() => {
+  specVoucherCancelation = pactum.spec();
+});
+
+// Scenario: The user successfully cancels a voucher
+Given('The user wants to cancel a voucher with voucher serial number 1234', () => {
+  serialNumberParam = '1234';
+  return serialNumberParam;
+});
+
+When('The user triggers an action with a valid payload to cancel a voucher', () => {
+  requestFunction();
+});
+
+Then('The user successfully cancels the voucher', async () => {
+  await specVoucherCancelation.toss();
+  specVoucherCancelation.response().should.have.status(200);
+  specVoucherCancelation.response().should.have.jsonLike({
+    'message': 'Voucher cancelled'
+  });
+});
+
+// Scenario: The user is not able to cancel a voucher with voucher serial number 4321
+Given('The user wants to cancel a voucher with voucher serial number 4321', () => {
+  serialNumberParam = '4321';
+  return serialNumberParam;
+});
+
+When('The user triggers an action without payload to cancel a voucher', () => {
+  specVoucherCancelation.patch(baseUrl).withBody()
```

Comment:
I see that in this test case you don't use requestFunction. 
Can you take same approach with setting up body value that you take with  `voucherserialnumber` query param? Body variable would be defined in `@Given` statement and used by requestFunction accordingly.

Reply:
Good idea!

Reply:
@mgrochalskisoldevelo @dborowiecki can we close this thread? Or still need to add some changes?

Reply:
@KarolinaKopacz  - In my opinion we can close it. Each test case uses now requestFunction with body variable defined in ```@Given``` statement